express-notes-api/server.ts

```diff
@@ -0,0 +1,133 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+type Note = { id: number; content: string };
+type Notes = Record<number, Note>;
+type Data = { nextId: number; notes: Notes };
+
+const app = express();
+
+app.use(express.json());
+
+// Handles GET request for all notes.
+app.get('/api/notes', async (req, res) => {
+  try {
+    const data = await JSON.parse(await readFile('data.json', 'utf8'));
+    const notes: Notes = data.notes;
+    const notesArr = [];
+    for (const key in notes) {
+      notesArr.push(notes[key]);
+    }
+    res.json(notesArr);
+  } catch (err) {
+    console.log('There was an error:', err);
```

Comment:
You need to send the client an error response in this case. Same in your other catch blocks.